dartsim/src/JointFeatures.cc

```diff
@@ -324,6 +326,94 @@ void JointFeatures::SetJointMaxEffort(
   joint->setForceUpperLimit(_dof, _value);
 }
 
+/////////////////////////////////////////////////
+
+void JointFeatures::SetJointFriction(
+       const Identity &_id, std::size_t _dof, double _value)
+{
+  auto joint = this->ReferenceInterface<JointInfo>(_id)->joint;
+   
+  // Take extra care that the value is valid. A nan can cause the DART
+  // constraint solver to fail, which will in turn either cause a crash or
+  // collisions to fail
+
+  if(std::isnan(_value))
+  {
+    gzerr << "Invalid joint friction value [" << _value
+           << "] commanded on joint [" << joint->getName() << " DOF " << _dof
+           << "]. The command will be ignored\n";
+    return;
+  }
+  
+  joint->setCoulombFriction(_dof, _value);
+}
+
+/////////////////////////////////////////////////
+
+void JointFeatures::SetJointDampingCoefficient(
+       const Identity &_id, std::size_t _dof, double _value)
+{
+  auto joint = this->ReferenceInterface<JointInfo>(_id)->joint;
+   
+  // Take extra care that the value is valid. A nan can cause the DART
+  // constraint solver to fail, which will in turn either cause a crash or
+  // collisions to fail
+
+  if(std::isnan(_value))
+  {
+    gzerr << "Invalid joint damping value [" << _value
+           << "] commanded on joint [" << joint->getName() << " DOF " << _dof
+           << "]. The command will be ignored\n";
+    return;
+  }
+  
+  joint->setDampingCoefficient(_dof, _value);
+}
+
+/////////////////////////////////////////////////
+
+void JointFeatures::SetJointSpringStiffness(
+       const Identity &_id, std::size_t _dof, double _value)
+{
+  auto joint = this->ReferenceInterface<JointInfo>(_id)->joint;
+   
+  // Take extra care that the value is valid. A nan can cause the DART
+  // constraint solver to fail, which will in turn either cause a crash or
+  // collisions to fail
+
+  if(std::isnan(_value))
+  {
+    gzerr << "Invalid joint spring stiffness value [" << _value
+           << "] commanded on joint [" << joint->getName() << " DOF " << _dof
+           << "]. The command will be ignored\n";
+    return;
+  }
+  
+  joint->setSpringStiffness(_dof, _value);
+}
+
+/////////////////////////////////////////////////
+
+void JointFeatures::SetJointRestPosition(
+       const Identity &_id, std::size_t _dof, double _value)
+{
+  auto joint = this->ReferenceInterface<JointInfo>(_id)->joint;
+   
+  // Take extra care that the value is valid. A nan can cause the DART
+  // constraint solver to fail, which will in turn either cause a crash or
+  // collisions to fail
+
+  if(std::isnan(_value))
```

Comment:
nit: add space after `if`. There are a few of them in this file.

```suggestion
  if (std::isnan(_value))
```